cartographer_ros/cartographer_ros/offline_node_main.cc

```diff
@@ -124,6 +129,16 @@ void Run(const std::vector<string>& bag_filenames) {
     static_tf_broadcaster.sendTransform(urdf_transforms);
   }
 
+  ros::AsyncSpinner async_spinner(kSingleThreaded);
+  async_spinner.start();
+  rosgraph_msgs::Clock clock;
+  auto clock_republish_timer = node.node_handle()->createWallTimer(
```

Comment:
Thanks for the explanation. Good for now, but in general let's try to keep code reviews focused and move further feature work to separate PRs.